### DISCUSSION TITLE
Regarding the definition of new material

### DISCUSSION THREAD

**Discussion by manikumd22** [idaholab/moose#26306] [Q&A Modules: Solid mechanics]
I'm working on multiscale modeling and want to input the properties such as stress tensor from outside. As in FEM, we calculate the tangent stiffness matrix (K_T) which consists of two terms K_geometric (function of stress) and K_material(function of material constants) and internal force(function of stress). This stress tensor (input from outside) will evaluate the tangent stiffness matrix and internal force at every iteration.

I want to define a new material that will take deformation tensor (F) as input from Moose and give stress tensor (S) as output back to MOOSE which will be used for the evaluation of stiffness, the code for this calculation from F to S is based on small scale and I already coded on C++ framework

I am struggling to do that

<img width="805" alt="Screenshot 2023-12-12 at 10 21 14 PM" src="https://github.com/idaholab/moose/assets/104883778/068a142d-71fc-4a40-90e5-ad1c6bdee98e">

**Comment by GiudGiud** [comment]
Hello

This is more of a tensor mechanics question, I m reclassifying this to get it seen by the relevant people

Guillaume
